broker/src/main/java/io/camunda/zeebe/broker/partitioning/PartitionManager.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.partitioning;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+
+import io.atomix.cluster.ClusterMembershipService;
+import io.atomix.cluster.messaging.ClusterCommunicationService;
+import io.atomix.primitive.partition.ManagedPartitionGroup;
+import io.atomix.primitive.partition.ManagedPartitionService;
+import io.atomix.primitive.partition.impl.DefaultPartitionService;
+import io.atomix.raft.partition.RaftPartitionGroup;
+import io.atomix.raft.partition.RaftPartitionGroup.Builder;
+import io.atomix.utils.concurrent.Futures;
+import io.camunda.zeebe.broker.clustering.atomix.AtomixFactory;
+import io.camunda.zeebe.broker.system.configuration.BrokerCfg;
+import io.camunda.zeebe.broker.system.configuration.ClusterCfg;
+import io.camunda.zeebe.broker.system.configuration.DataCfg;
+import io.camunda.zeebe.broker.system.configuration.NetworkCfg;
+import io.camunda.zeebe.logstreams.impl.log.ZeebeEntryValidator;
+import io.camunda.zeebe.snapshots.ReceivableSnapshotStoreFactory;
+import io.camunda.zeebe.util.FileUtil;
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import org.agrona.IoUtil;
+
+public class PartitionManager {
+  protected volatile CompletableFuture<Void> closeFuture;
+  private final ManagedPartitionService partitions;
+  private final ManagedPartitionGroup partitionGroup;
+
+  protected PartitionManager(
+      final BrokerCfg configuration,
+      final ReceivableSnapshotStoreFactory snapshotStoreFactory,
+      final ClusterMembershipService membershipService,
+      final ClusterCommunicationService communicationService) {
+
+    partitionGroup = buildRaftPartitionGroup(configuration, snapshotStoreFactory);
+
+    partitions = buildPartitionService(membershipService, communicationService);
+  }
+
+  // TODO remove static modifier
```

Comment:
:question: What's the plan for this later? Will it stay in the partition manager? Will we have a partition group builder (well not builder, that's an overloaded word, but you get what I mean)?

Reply:
Outdated